elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryConstructor.java

```
@@ -261,29 +308,36 @@ private String constructTableOrSubselect(Class<?> cls) {
 
     /**
      * Given a list of columns to sort on, constructs an ORDER BY clause in SQL.
-     *
-     * @param entityClass The class to sort.
      * @param sortClauses The list of sort columns and their sort order (ascending or descending).
      * @return A SQL expression
      */
-    private String extractOrderBy(Class<?> entityClass, Map<Path, Sorting.SortOrder> sortClauses) {
+    private String extractOrderBy(Map<Path, Sorting.SortOrder> sortClauses, SQLQueryTemplate template) {
         if (sortClauses.isEmpty()) {
             return "";
         }
 
         //TODO - Ensure that order by columns are also present in the group by.
+
         return " ORDER BY " + sortClauses.entrySet().stream()
                 .map((entry) -> {
-                    Path path = entry.getKey();
-                    path = expandJoinToPath(path);
+                    Path expandedPath = expandJoinToPath(entry.getKey());
                     Sorting.SortOrder order = entry.getValue();
 
-                    Path.PathElement last = path.lastElement().get();
+                    Path.PathElement last = expandedPath.lastElement().get();
+
+                    MetricFunctionInvocation metric = template.getMetrics().stream()
+                            // TODO: filter predicate should support alias
+                            .filter(invocation -> invocation.getAlias().equals(last.getFieldName()))
+                            .findFirst()
+                            .orElse(null);
```

Throw exception instead of return null?